txpool/txpool_test.go

```diff
@@ -43,3 +43,147 @@ func TestPoolFlexibility(t *testing.T) {
 		t.Fatal("pool did not reject very outdated transaction")
 	}
 }
+
+func TestSpendInvalidEphemeralOutput(t *testing.T) {
```

Comment:
these tests seem more appropriate for `consensus` than `txpool`. Maybe create `consensus/validation_test.go`. I don't think we even need `ChainSim` for a test like this; `ChainSim` is more for dealing with multiple chains/forks. It might be helpful to define *some* helper functions/types though; there's a lot of boilerplate in e.g. `scratch_test.go` that I'd prefer not to replicate in every consensus test.

Reply:
How would you feel about splitting the tests? One for `txpool` for the first subtest, and the other for `consensus` to test `ValidationContext.ValidateTransactionSet`. I combined them to avoid duplicating the transaction setup, but I agree they don't completely fit in `txpool`. 

I think a single helper to set up a genesis context will get rid of the need for `ChainSim` in these tests, looking at `TestScratchChain` it could use a `MineBlockWithTxns` and `SignTxn` helper.

Reply:
I'm not seeing the benefit of the txpool tests -- seems like they're just indirectly testing `consensus.ValidateTransaction`  and `consensus.ValidateBlock`?

I suppose there's some value in testing that the txpool can link together transactions that depend on each other, but that should be a separate test.

Reply:
There's separate ephemeral output validation code in the txpool:
https://github.com/SiaFoundation/sunyata/blob/7b6147be07454c91e79808d3207d5b79153f8c7e/txpool/txpool.go#L40

Reply:
True!